libs/aio/odc/aio/__init__.py

```diff
@@ -557,12 +558,19 @@ def do_dir_query(qq):
             stream = s3.find(qq.base, pred=lambda o: o.url.endswith(postfix), **kw)
     else:
         # fixed depth query
+        _, prefix = s3_url_parse(glob_pattern)
+        dirs_glob = '/'.join(prefix.split('/')[:-1])
```

Comment:
since you split it back to list anyway, maybe no point putting it back into a string?

Reply:
I actually meant `dirs_glob`, no point merging it back into a string if you only use it as a list inside `dirs_pred`
```python
dirs_glob = prefix.split('/')[:-1]
...
    _glob = '/'.join(dirs_glob[:n]) + '/'
```

Reply:
yeah good point, I've fixed this up. I'll rebase and merge 👍